gapic/schema/wrappers.py

```diff
@@ -866,13 +866,19 @@ def paged_result_field(self) -> Optional[Field]:
         """Return the response pagination field if the method is paginated."""
         # If the request field lacks any of the expected pagination fields,
         # then the method is not paginated.
-        for page_field in ((self.input, int, 'page_size'),
-                           (self.input, str, 'page_token'),
+        for page_field_token in ((self.input, str, 'page_token'),
                            (self.output, str, 'next_page_token')):
-            field = page_field[0].fields.get(page_field[2], None)
-            if not field or field.type != page_field[1]:
+            field = page_field_token[0].fields.get(page_field_token[2], None)
+            if not field or field.type != page_field_token[1]:
                 return None
```

Comment:
Great extension. Since we're changing things here, can we tweak the loop so that we're destructuring into names instead of indexing the tuple?

```python
for source, name in ((self.input, "page_token"), (self.output, "next_page_token")):
```
Something like that

Reply:
Good idea. Done